dependencies.gradle

```diff
@@ -48,6 +48,7 @@ ext {
             ],
             junit                : "junit:junit:$junitVersion",
             kotlin               : "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion",
+            kotlinReflect        : "org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion",
```

Comment:
Are we using it? Might be worth removing usages

Reply:
We use it in quite a few places for `KClass`. Maybe we can change those to just `Class`?

Reply:
We'd have to also change calls from `::class` to `::class.java`, so it would be a breaking change in quite a few places.

Reply:
As far as I know, that doesn't require full reflect artifact. After looking through dependencies, `mockito` is the culprit. Adding `kotlinReflect` to `testImplementation` fixes the issue.

Reply:
Good catch, updated the PR to reflect this. No pun intended 😛